resources/dw/virtualwarehouse/impala/resource_impala_vw.go

```diff
@@ -0,0 +1,262 @@
+	tflog.Warn(ctx, "Read operation is not implemented yet.")
+}
+
+func (r *impalaResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	tflog.Warn(ctx, "Update operation is not implemented yet.")
+}
+
+func (r *impalaResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state resourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	clusterID := state.ClusterID.ValueStringPointer()
+	vwID := state.ID.ValueStringPointer()
+
+	if clusterID == nil || vwID == nil {
+		resp.Diagnostics.AddError(
+			"Invalid State",
+			"ClusterID or Virtual Warehouse ID is missing.",
+		)
+		return
+	}
+
+	err := r.deleteVirtualWarehouse(ctx, clusterID, vwID)
+	if err != nil {
+		if strings.Contains(err.Error(), "Virtual Warehouse not found") {
+			return
+		}
+		resp.Diagnostics.AddError(
+			"Error deleting Impala Virtual Warehouse",
+			"Could not delete Impala Virtual Warehouse, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	// Handle polling for not async call
+	pollingOptionsAvailable := state.PollingOptions != nil
+	isSynchronous := pollingOptionsAvailable && !state.PollingOptions.Async.ValueBool()
+	if isSynchronous {
+		err = r.pollForDeletion(ctx, state, clusterID, vwID)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error waiting for Impala Virtual Warehouse deletion",
+				"Could not delete Impala Virtual Warehouse, unexpected error: "+err.Error(),
+			)
+		}
+	}
+}
+
+func (r *impalaResource) stateRefresh(ctx context.Context, clusterID *string, vwID *string, callFailedCount *int, callFailureThreshold int) func() (any, string, error) {
+	return func() (any, string, error) {
+		tflog.Debug(ctx, "About to describe Impala")
+		params := operations.NewDescribeVwParamsWithContext(ctx).
+			WithInput(&models.DescribeVwRequest{ClusterID: clusterID, VwID: vwID})
+		resp, err := r.client.Dw.Operations.DescribeVw(params)
+		if err != nil {
+			if strings.Contains(err.Error(), "Virtual Warehouse not found") {
+				return &models.DescribeVwResponse{}, "Deleted", nil
+			}
+			*callFailedCount++
+			if *callFailedCount <= callFailureThreshold {
+				tflog.Warn(ctx, fmt.Sprintf("could not describe Data Warehouse Impala Virtual Warehouse "+
+					"due to [%s] but threshold limit is not reached yet (%d out of %d).", err.Error(), callFailedCount, callFailureThreshold))
+				return nil, "", nil
+			}
+			tflog.Error(ctx, fmt.Sprintf("error describing Data Warehouse Impala Virtual Warehouse due to [%s] "+
+				"failure threshold limit exceeded.", err.Error()))
+			return nil, "", err
+		}
+		*callFailedCount = 0
+		vw := resp.GetPayload()
+		tflog.Debug(ctx, fmt.Sprintf("Described Impala %s with status %s", vw.Vw.ID, vw.Vw.Status))
+		return vw, vw.Vw.Status, nil
+	}
+}
+
+func createVwRequestFromPlan(plan *resourceModel) *models.CreateVwRequest {
```

Comment:
why not make this a method on the struct? why a function?
```
func (plan *resourceModel) createVwRequestFromPlan() *models.CreateVwRequest {
```

Reply:
I don't have a strong opinion either ways but keeping it in the current format paves the way for reusability across hive, impala, trino and maybe DBC as well. While binding them to the structs would make them not usable but lemme know if you feel strongly towards any and would use that.

Reply:
Impala and Hive and Trino will have very different properties. You haven't added all of them yet, so these are common, but the majority is different. Aways try to group the functions and objects based on their responsibilities. You have a plan, with the type resourceModel. This resourceModel you will convert to a CreateVwRequest, and when you get back the properties from DescribeVw, you will store the properties in a resourceModel. Makes sense to add methods to resourceModel to handle the object transformations.